geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/MetadataFileSystemStorage.scala

```diff
@@ -127,48 +127,52 @@ abstract class MetadataFileSystemStorage(conf: Configuration,
   override def compact(partition: String, threads: Int): Unit = {
     val toCompact = getFilePaths(partition).asScala
 
-    val leaf = metadata.getPartitionScheme.isLeafStorage
-    val dataPath = StorageUtils.nextFile(metadata.getRoot, partition, leaf, extension, FileType.Compacted)
+    if (toCompact.lengthCompare(2) < 0) {
+      logger.debug(s"Skipping compaction for single data file: ${toCompact.mkString(", ")}")
```

Comment:
Hmmm... seems like we wouldn't need the .mkString for one file:)?

Reply:
could also be 0 files... it was either that or `toCompact.headOption.map(_.toString).getOrElse("")`